pkg/jobs/job_scheduler.go

```diff
@@ -135,23 +164,76 @@ func (s *jobScheduler) processSchedule(
 		return err
 	}
 
-	executor, err := NewScheduledJobExecutor(schedule.ExecutorType())
+	executor, err := s.lookupExecutor(schedule.ExecutorType())
 	if err != nil {
 		return err
 	}
 
 	// Grab job executor and execute the job.
+	log.Infof(ctx,
+		"Starting job for schedule %d (%s); next run scheduled for %s",
+		schedule.ScheduleID(), schedule.ScheduleName(), schedule.NextRun())
+
 	if err := executor.ExecuteJob(ctx, s.JobExecutionConfig, s.env, schedule, txn); err != nil {
 		return errors.Wrapf(err, "executing schedule %d", schedule.ScheduleID())
 	}
 
+	stats.started++
+
 	// Persist any mutations to the underlying schedule.
 	return schedule.Update(ctx, s.InternalExecutor, txn)
 }
 
+func (s *jobScheduler) lookupExecutor(name string) (ScheduledJobExecutor, error) {
+	if ex, ok := s.executors[name]; ok {
+		return ex, nil
+	}
+	ex, err := NewScheduledJobExecutor(name)
+	if err != nil {
+		return nil, err
+	}
+	if m := ex.Metrics(); m != nil {
+		s.registry.AddMetricStruct(m)
+	}
+	return ex, nil
+}
+
+func newLoopStats(
```

Comment:
I'm somewhat skeptical these are actually worth collecting / maintaining by default, but we can always remove them after 20.2 if we found we never used them, so I'm no opposed to adding them for now. I just might leave a TODO to come back and see if we ever actually acted on these or if we could avoid these extra scan / timeseries overhead/downsamples/etc.